devicetypes/smartenit/smartenit-zigbee-metering-outlet.src/smartenit-zigbee-metering-outlet.groovy

```diff
@@ -0,0 +1,134 @@
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	log.debug "event: ${event}"
+
+	if (event) {
+		if (event.name == "power") {
+			event = createEvent(name: event.name, value: (event.value as Integer), descriptionText: '{{ device.displayName }} power is {{ value }} Watts', translatable: true)
+		} else if (event.name == "switch") {
+			def descriptionText = event.value == "on" ? '{{ device.displayName }} is On' : '{{ device.displayName }} is Off'
+			event = createEvent(name: event.name, value: event.value, descriptionText: descriptionText, translatable: true)
+		}
+	} else {
+		def cluster = zigbee.parse(description)
+		log.debug "cluster def: ${cluster}"
+		def mapDescription = zigbee.parseDescriptionAsMap(description)
+
+		if (cluster && cluster.clusterId == zigbee.ONOFF_CLUSTER && cluster.command == ReportingResponse) {
+			if (cluster.data[0] == 0x00) {
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+				event = createEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+				event = null
+			}
+		} else if (mapDescription && (mapDescription.clusterInt == zigbee.SIMPLE_METERING_CLUSTER)) {
+			if (mapDescription.attrInt == MeteringCurrentSummation) {
+				event = createEvent(name: "energy", value: getFPoint(mapDescription.value)/EnergyDivisor)
+			} else if (mapDescription.attrInt == MeteringInstantDemand) {
+				event = createEvent(name: "power", value: getFPoint(mapDescription.value)/EnergyDivisor)
+			} else {
+				log.debug "Could not find attribute mapping for ${mapDescription.clusterInt} ${mapDescription.attrInt}"
+			} 
+		} else if (mapDescription && (mapDescription.clusterInt == ElectricalMeasurement)) {
+			if (mapDescription.attrInt == Voltage) {
+				event = createEvent(name: "voltage", value: getFPoint(mapDescription.value))
+			} else if (mapDescription.attrInt == Current) {
+				event = createEvent(name: "current", value: getFPoint(mapDescription.value)/CurrentDivisor, unit: "A")
+			}
+		} else {
+			log.warn "DID NOT PARSE MESSAGE for description : $description"
+			log.debug "${cluster}"
+		}
+	}
+	return event ? createEvent(event) : event
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	return zigbee.onOffRefresh()
+}
+
+def refresh() {
+	zigbee.onOffRefresh() +
+	zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, MeteringCurrentSummation) + 
+	zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER , Voltage) + 
+	zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER , Current) +
+	zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER , PowerMultiplier) +
+	zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER , PowerDivisor) + 
```

Comment:
I see these are being read, but I don't see where the returned values are being stored.